### ISSUE TITLE
StatsD streaming does not work via TCP

### ISSUE THREAD

**Issue by CitizenUnderdog** [F5Networks/f5-telemetry-streaming#45]
### Environment
 * Telemetry Streaming Version: 1.10.0
 * BIG-IP Version: 13.1.3.2.0.0.4

### Summary
When Telemetry Consumer is configured as StatsD with TCP protocol, the F5 emits metrics in UDP protocol instead.

### Steps To Reproduce
1. Submit the following declaration:
```json
{
    "class": "Telemetry",
    "My_System": {
        "class": "Telemetry_System",
        "systemPoller": {
            "interval": 60
        }
    },
    "My_Listener": {
        "class": "Telemetry_Listener",
        "port": 6514
    },
    "Statsd_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Statsd",
        "host": "10.57.152.66",
        "protocol": "tcp",
        "port": 8125
    }
}
```

2. Observe the following  response:

Configuration is accepted 200 OK:
```json
{
    "class": "Telemetry",
    "My_System": {
        "class": "Telemetry_System",
        "systemPoller": {
            "interval": 60
        }
    },
    "My_Listener": {
        "class": "Telemetry_Listener",
        "port": 6514
    },
    "Statsd_Consumer": {
        "class": "Telemetry_Consumer",
        "type": "Statsd",
        "host": "10.57.152.66",
        "protocol": "tcp",
        "port": 8125
    }
}```


However packet captures reveal UDP is still protocol transmitted:
```
[COF5ADMN19@ttbnlbpoc82a-mgmt:/S1-green-P::Standby:In Sync] ~ # tcpdump -nni 0.0 host 10.57.152.66
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on 0.0, link-type EN10MB (Ethernet), capture size 65535 bytes
15:24:11.134283 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135499 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135502 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135503 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135504 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 98 out slot1/tmm0 lis=
15:24:11.135505 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 99 out slot1/tmm0 lis=
15:24:11.135514 IP 10.57.152.66 > 10.63.92.7: ICMP 10.57.152.66 udp port 8125 unreachable, length 132 in slot1/tmm0 lis=
```
### Expected Behavior
We expected TCP protocol to be used for transmission

### Actual Behavior
Packet captures show UDP is used instead of TCP. Even after a reboot of the F5 device.

```
[COF5ADMN19@ttbnlbpoc82a-mgmt:/S1-green-P::Standby:In Sync] ~ # tcpdump -nni 0.0 host 10.57.152.66
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on 0.0, link-type EN10MB (Ethernet), capture size 65535 bytes
15:24:11.134283 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135499 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135502 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135503 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 96 out slot1/tmm0 lis=
15:24:11.135504 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 98 out slot1/tmm0 lis=
15:24:11.135505 IP 10.63.92.7.42522 > 10.57.152.66.8125: UDP, length 99 out slot1/tmm0 lis=
15:24:11.135514 IP 10.57.152.66 > 10.63.92.7: ICMP 10.57.152.66 udp port 8125 unreachable, length 132 in slot1/tmm0 lis=
```

**Comment by CitizenUnderdog** [comment]
Looks like the issue might be related to this?
https://github.com/F5Networks/f5-telemetry-streaming/blob/556cd01de7686fb03b69c603771c3ff624eac726/src/lib/consumers/Statsd/index.js#L11

node-statsd only supports UPD?

**Comment by dstokesf5** [comment]
As @CitizenUnderdog pointed out, the StatsD consumer only works with UDP currently. I have created AUTOTOOL-1500 to update the schema to make this clearer.

**Comment by dstokesf5** [comment]
The next release (1.13.0) will support using UDP with the StatsD consumer.

**Comment by murphymj25** [comment]
Is there any plans to support TCP for statsd?

**Comment by dstokesf5** [comment]
Sorry, mistyped that. Meant to say it will support TCP in 1.13.0.

**Comment by murphymj25** [comment]
Thanks for the clarification!

**Comment by dstokesf5** [comment]
This was resolved in v1.12.0. Closing this issue.
